torch/_dynamo/guards.py

```diff
@@ -493,6 +494,15 @@ def DETERMINISTIC_ALGORITHMS(self, guard: Guard):
             code = "not ___are_deterministic_algorithms_enabled()"
         self._produce_guard_code(guard, [code])
 
+    def TORCH_FUNCTION_STATE(self, guard: Guard):
+        assert guard.source is GuardSource.GLOBAL
+        code = None
+        if convert_frame.initial_torch_function_state:
+            code = "___is_torch_function_enabled()"
+        else:
+            code = "not ___is_torch_function_enabled()"
+        self._produce_guard_code(guard, [code])
```

Comment:
Need to improve the cpp bindings, and then improve this guard. TorchFunctionDisabledState is actually a ternary state:

```
enum TorchFunctionDisabledState { ENABLED, SUBCLASSES_DISABLED, ALL_DISABLED };
```

Should match this exactly.

Reply:
More generally, we're getting a lot of one off "state" guards. We should have a single, consolidated guard, which checks all of these "must match" guards in C++.

Reply:
Lol this is my offline comment. I'll do this, I don't want to block @mlazos on it.